frontend/src/app/pages/exercises/exercise/shared/exercise-map/utility/simulated-region-modify.ts

```diff
@@ -0,0 +1,69 @@
+import type { Feature } from 'ol';
+import { primaryAction, shiftKeyOnly } from 'ol/events/condition';
+import type { LineString, Point } from 'ol/geom';
+import { Modify } from 'ol/interaction';
+import type VectorLayer from 'ol/layer/Vector';
+import type VectorSource from 'ol/source/Vector';
+import type { CornerName, ModifyGeometry } from './modify-helper';
+
+const originCornerDict: {
+    [key in CornerName]: { originIndex: number; cornerIndex: number };
+} = {
+    bottomRight: { originIndex: 0, cornerIndex: 2 },
+    bottomLeft: { originIndex: 1, cornerIndex: 3 },
+    topLeft: { originIndex: 2, cornerIndex: 0 },
+    topRight: { originIndex: 3, cornerIndex: 1 },
+};
+
+export function createSimulatedRegionModify(
+    simulatedRegionLayer: VectorLayer<VectorSource<LineString>>
+): Modify {
+    const defaultStyle = new Modify({
+        source: simulatedRegionLayer.getSource()!,
+    })
+        .getOverlay()
+        .getStyleFunction();
+    return new Modify({
+        source: simulatedRegionLayer.getSource()!,
+        condition: (event) => primaryAction(event) && shiftKeyOnly(event),
+        deleteCondition: () => false,
+        insertVertexCondition: () => false,
+        style: (feature) => {
+            (feature.get('features') as Feature[]).forEach((modifyFeature) => {
+                const modifyGeometry = modifyFeature.get(
+                    'modifyGeometry'
+                ) as ModifyGeometry;
+                if (!modifyGeometry) {
+                    return;
+                }
+                const mouseCoordinate = (
+                    feature.getGeometry() as Point
+                ).getCoordinates();
+
+                const corners = modifyGeometry.geometry.getCoordinates();
+                const { originIndex, cornerIndex } =
+                    originCornerDict[modifyGeometry.modifyCorner];
+                // The corners array must be big enough.
+                if (
+                    corners.length <= originIndex ||
+                    corners.length <= cornerIndex
+                ) {
+                    throw new Error(
+                        `corners must be at least as big to satisfy originIndex (${originIndex}) and cornerIndex (${cornerIndex}): ${corners}`
+                    );
+                }
+                const origin = corners[originIndex]!;
+                const corner = corners[cornerIndex]!;
+                modifyGeometry.geometry.scale(
+                    (mouseCoordinate[0]! - origin[0]!) /
+                        (corner[0]! - origin[0]!),
+                    (origin[1]! - mouseCoordinate[1]!) /
+                        (origin[1]! - corner[1]!),
+                    origin
+                );
+            });
+            // This renders the default LineString style. The function ignores all arguments.
+            return defaultStyle!(feature, 123);
```

Comment:
If the function ignores all arguments, why do we provide any?

Reply:
Because the typings of OpenLayers are incorrect here and require an unused second argument. I agree that the comment should make this clearer. Maybe a `// @ts-expect-error` could be used?

Reply:
I'd also though about using `// @ts-expect-error`, but what I don't like about this solution is that (as far as I know) we cannot specify *what* error to expect, so possible future errors or a change in the type definition which might resolve this issue would not be detected/reported by TypeScript.

Therefore, I'd rather just make the comment more clear than using `// @ts-expect-error`.

Reply:
I've updated the comment